src/metabase/models/native_query_snippet.clj

```diff
@@ -99,10 +100,21 @@
     []))
 
 (defmethod serdes/storage-path "NativeQuerySnippet" [snippet ctx]
-  ;; Intended path here is ["snippets" "nested" "collections" "snippet_eid_and_slug"]
+  ;; Intended path here is ["snippets" "<nested ... collections>" "<snippet_eid_and_slug>"]
   ;; We just the default path, then pull it apart.
-  ;; The default is ["collections" "nested" collections" "nativequerysnippets" "base_name"]
+  ;; The default is ["collections" "<nested ... collections>" "nativequerysnippets" "<base_name>"]
   (let [basis  (serdes/storage-default-collection-path snippet ctx)
         file   (last basis)
         colls  (->> basis rest (drop-last 2))] ; Drops the "collections" at the start, and the last two.
     (concat ["snippets"] colls [file])))
+
+
+(defmethod serdes/load-one! "NativeQuerySnippet" [ingested maybe-local]
+  ;; if we got local snippet in db and it has same name as incoming one, we can be sure
+  ;; there will be no conflicts and skip the query to the db
+  (if (and (not= (:name ingested) (:name maybe-local))
+           (t2/exists? :model/NativeQuerySnippet
+                       :name (:name ingested) :entity_id [:!= (:entity_id ingested)]))
+    (recur (update ingested :name str " (copy)")
+           maybe-local)
```

Comment:
is there any benefit to using recur here instead of calling `serdes/default-load-one!` directly?

Reply:
yeah, the test checks for the second name conflict — ` (copy) (copy)` — exactly for that case.